### PR TITLE
Replace Checksum with proto equivalent

### PR DESCRIPTION
**Pull request by alexshtin**
**What changed?**
`checksum.Checksum` struct was replaced with `persistenceblobs.Checksum` from proto.

**Why?**
This is next step to complete `persistence.WorkflowMutableState` migration to proto to get rid of unnecessary conversion between go and proto struct.

**How did you test it?**
Run all tests.

**Potential risks**
No risks.

### CODE REVIEW

File: common/persistence/dataInterfaces.go [hunk 1]
```diff
@@ -467,7 +466,7 @@ type (
 		ExecutionStats      *persistenceblobs.ExecutionStats
 		BufferedEvents      []*historypb.HistoryEvent
 		VersionHistories    *VersionHistories
-		Checksum            checksum.Checksum
+		Checksum            persistenceblobs.Checksum
```
Comment (alexshtin) [inline common/persistence/dataInterfaces.go:469]: After merging with WEI, nothing will block from converting entire WMS to proto.

Comment (sergeybykov) [reply common/persistence/dataInterfaces.go:469]: Will the checksum value stay the same? If not, will this break backward compatibility?

Comment (alexshtin) [reply common/persistence/dataInterfaces.go:395]: Yes, proto struct was already used for persistence. But before they were converted (basic 1:1 mapping) to go struct, go struct was used in code, and then converted back to proto for persistence. This PR just removes this mapping and change all internal code to use proto struct, not go struct.

Comment (alexshtin) [reply common/persistence/dataInterfaces.go:395]: They way how we compute crc32 stays the same.

**Review by samarabbas — APPROVED** [review]

**Pull request merged**